rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -924,6 +924,73 @@ void saveFederationElection_shouldSaveInStorage() {
         assertArrayEquals(federationElectionSerializedWithVote, actualAbiCallElectionSerialized);
     }
 
+    @Nested
+    @Tag("proposed federation tests")
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    class ProposedFederationTests {
+        private final ActivationConfig.ForBlock preLovellActivations = ActivationConfigsForTest.arrowhead631().forBlock(0L);
+        private final ActivationConfig.ForBlock lovellActivations = ActivationConfigsForTest.lovell700().forBlock(0L);
+        private final Federation proposedFederation = new P2shErpFederationBuilder().build();
+        private StorageAccessor bridgeStorageAccessor;
+        private FederationStorageProvider federationStorageProvider;
+
+        @BeforeEach
+        void setUp() {
+            bridgeStorageAccessor = new InMemoryStorage();
+            federationStorageProvider = new FederationStorageProviderImpl(bridgeStorageAccessor);
+        }
+
+        @Test
+        void saveProposedFederation_whenProposedFederationIsNotSet_shouldNotSave() {
+            federationStorageProvider.save(networkParameters, lovellActivations);
+
+            assertNull(getProposedFederationFromRepository());
+        }
+
+        @Test
+        void saveProposedFederation_preRSKIP419_whenProposedFederationIsSet_shouldNotSave() {
+            federationStorageProvider.setProposedFederation(proposedFederation);
+            federationStorageProvider.save(networkParameters, preLovellActivations);
+
+            assertNull(getProposedFederationFromRepository());
+        }
+
+        @Test
+        void saveProposedFederation_whenProposedFederationIsSet_shouldSave() {
+            federationStorageProvider.setProposedFederation(proposedFederation);
+            federationStorageProvider.save(networkParameters, lovellActivations);
+
+            assertEquals(proposedFederation, getProposedFederationFromRepository());
```

Comment:
Should we also assert that the correct storage version is stored?

Reply:
done